restapi.go

```diff
@@ -2339,3 +2361,215 @@ func (s *Session) RelationshipsMutualGet(userID string) (mf []*User, err error)
 	err = unmarshal(body, &mf)
 	return
 }
+
+// ------------------------------------------------------------------------------------------------
+// Functions specific to application (slash) commands
+// ------------------------------------------------------------------------------------------------
+
+// ApplicationCommandCreate creates an global application command and returns it.
+// appID       : The application ID. If empty - s.State.User.ID. Note: works only with opened session.
+// guildID     : Guild ID to create guild-specific application command. If empty - creates global appication command.
+// cmd         : New application command data.
+func (s *Session) ApplicationCommandCreate(appID string, guildID string, cmd *ApplicationCommand) (rcmd *ApplicationCommand, err error) {
+	if appID == "" {
+		appID = s.State.User.ID
+	}
+
+	endpoint := EndpointApplicationGlobalCommands(appID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommands(appID, guildID)
+	}
+
+	var body []byte
+	body, err = s.RequestWithBucketID("POST", endpoint, *cmd, endpoint)
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &rcmd)
+
+	return
+}
+
+// ApplicationCommandEdit edits application command and returns old command data.
+// appID       : The application ID. If empty - s.State.User.ID. Note: works only with opened session.
+// cmdID       : Application command ID to edit.
+// guildID     : Guild ID to edit guild-specific application command. If empty - edits global application command.
+// cmd         : Updated application command data.
+func (s *Session) ApplicationCommandEdit(appID, cmdID, guildID string, cmd *ApplicationCommand) (oldcmd *ApplicationCommand, err error) {
+	if cmdID == "" {
+		return nil, errors.New("invalid command id")
+	}
+
+	if appID == "" {
+		appID = s.State.User.ID
+	}
+
+	endpoint := EndpointApplicationGlobalCommand(appID, cmdID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommand(appID, guildID, cmdID)
+	}
+
+	var body []byte
+	body, err = s.RequestWithBucketID("PATCH", endpoint, *cmd, endpoint)
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &cmd)
+
+	return
```

Comment:
```return &cmd, nil```

Reply:
No. Look at the answer on other comment.

Reply:
Naked returns significantly hurt readability. In fact, this is a prime example of that. The function defines the return data as `(oldcmd *ApplicationCommand, err error)`, creating `nil` as the zero value of `oldcmd`. Later in the function `oldcmd` never gets assigned a value so it will always be nil. Instead you probably want to return `&cmd, nil`